### PR TITLE
Added list_sibling_adapters() method to Adapter class

### PR DESCRIPTION
**Pull request by andy-maier**
For details, see the commit message.

Particular review points:
* Is "sibling" an appropriate term for the other adapters on the same card

### CODE REVIEW

File: tests/end2end/test_adapter.py [hunk 1]
```diff
@@ -467,3 +467,79 @@ def test_adapter_list_assigned_part(dpm_mode_cpcs):  # noqa: F811
...
+def base_adapter_id(adapter_id, family):
+    """
+    Return the base adapter_id of an adapter_id. Both are in hex.
+    """
+    # Adapter families that have no physical adapter card
+    logical_families = [
+        'hipersocket',
+    ]
+    pchid = int(adapter_id, 16)
+    if family in logical_families:
```
Comment (EdGue42) [inline tests/end2end/test_adapter.py:481]: your if assigns the same value for then/else ?

File: tests/end2end/test_adapter.py [hunk 2]
```diff
@@ -467,3 +467,79 @@ def test_adapter_list_assigned_part(dpm_mode_cpcs):  # noqa: F811
...
                 tmp_part.delete()
+
+
+def base_adapter_id(adapter_id, family):
+    """
+    Return the base adapter_id of an adapter_id. Both are in hex.
+    """
+    # Adapter families that have no physical adapter card
+    logical_families = [
+        'hipersocket',
```
Comment (simon-spinner) [inline tests/end2end/test_adapter.py:478]: What about ism and icp?

File: zhmcclient/_adapter.py [hunk 3]
```diff
@@ -679,3 +679,58 @@ def list_assigned_partitions(self, full_properties=False, filter_args=None):
...
+          :exc:`~zhmcclient.AuthError`
+          :exc:`~zhmcclient.ConnectionError`
+        """
+        # This algorithm is based on the fact that physical adapter cards
+        # have their PCHIDs always within a range of 4 adjacent PCHIDs that
+        # start at a multiple of 4.
+
+        self_pchid = int(self.prop('adapter-id'), 16)
+        if self_pchid >= int('700', 16):
+            # Hipersocket adapter
```
Comment (simon-spinner) [inline zhmcclient/_adapter.py:726]: just on your comment: could be also an ISM or ICP adapter (they also have VCHIDs >700).

File: tests/end2end/test_adapter.py (hunk 1 above)
Comment (andy-maier) [reply tests/end2end/test_adapter.py:481]: The code says:
```
    if family in logical_families:
        base_pchid = pchid
    else:
        base_pchid = pchid // 4 * 4
```
That is not the same value.

File: zhmcclient/_adapter.py (hunk 3 above)
Comment (andy-maier) [reply zhmcclient/_adapter.py:726]: With this code, I want to detect adapters without siblings, where sibling is defined as the set of other PCHIDs on the same same physical card. So I need to understand more details about ISM and ICP.

Comment (andy-maier) [reply zhmcclient/_adapter.py:726]: Changed that to "Virtual adapter" and added more explanatory comments.

File: tests/end2end/test_adapter.py (hunk 2 above)
Comment (andy-maier) [reply tests/end2end/test_adapter.py:478]: Added "ism" and "coupling" to the list of virtual families and renamed the variable to "virtual_families".

**Pull request merged**